### PR TITLE
Add ProxySQL to Load Balance MySQL Query Requests

### PR DESCRIPTION
**Pull request by shudipta**

### CODE REVIEW

File: apis/kubedb/v1alpha1/proxysql_types.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	"github.com/appscode/go/encoding/json/types"
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	mona "kmodules.xyz/monitoring-agent-api/api/v1"
+	ofst "kmodules.xyz/offshoot-api/api/v1"
+)
+
+const (
+	ResourceCodeProxySQL     = "psql"
```
Comment (tamalsaha) [inline apis/kubedb/v1alpha1/proxysql_types.go]: Is this the right short code? `psql` is more for postgresql

File: apis/kubedb/v1alpha1/proxysql_types.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+	// Mode specifies the type of MySQL/Percona-XtraDB/MariaDB cluster for which proxysql
+	// will be configured. It must be either "Galera" or "GroupReplication"
+	Mode *LoadBalanceMode `json:"mode,omitempty"`
+
+	// Backend specifies the information about backend MySQL/Percona-XtraDB/MariaDB servers
+	Backend *ProxySQLBackendSpec `json:"backend,omitempty"`
+
+	// StorageType can be durable (default) or ephemeral
+	StorageType StorageType `json:"storageType,omitempty"`
```
Comment (tamalsaha) [inline apis/kubedb/v1alpha1/proxysql_types.go]: Why do we need storage?

File: apis/kubedb/v1alpha1/proxysql_types.go (hunk 1 above)
Comment (shudipta) [reply apis/kubedb/v1alpha1/proxysql_types.go]: resolved

File: apis/kubedb/v1alpha1/proxysql_types.go (hunk 2 above)
Comment (shudipta) [reply apis/kubedb/v1alpha1/proxysql_types.go]: resolved

**Pull request merged**